p4-16/spec/P4-16-spec.mdk

```diff
@@ -6416,6 +6416,160 @@ ck16.update( { h.ipv4.ttl, h.ipv4.proto } );
 h.ipv4.hdrChecksum = ck16.get();
 ~ End P4Example
 
+
+# Appendix: Restrictions on compile time and run time calls
+
+This appendix summarizes restrictions on compile time and run time
+calls that can be made.  Many of them are described earlier in this
+document, but are collected here for easy reference.
+
+The stateful types of objects in P4~16~ are packages, parsers,
+controls, externs, and tables.  All other types are referred to as
+"value types" here.
+
+Some guiding principles:
+
+- Packages can be instantiated, but have no run-time behavior, so
```

Comment:
Understood.  That bullet point wasn't explaining very many of the later restrictions, so I'll just delete it.